web/src/main/java/com/epam/esm/controller/UserController.java

```diff
@@ -0,0 +1,100 @@
+        this.orderService = orderService;
+        this.userAssembler = assembler;
+        this.orderAssembler = orderAssembler;
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<UserDto> getById(@PathVariable long id) {
+        UserDto foundUser = userService.findById(id);
+        return ResponseEntity.ok(userAssembler.toModel(foundUser));
+    }
+
+    @GetMapping
+    public ResponseEntity<CollectionModel<UserDto>> getAllForQuery(WebRequest request,
+                                                                   @RequestParam(required = false, defaultValue = ApplicationConstants.DEFAULT_LIMIT + "") @Min(value = 0, message = "Limit parameter must be greater or equal 0") Integer limit,
+                                                                   @RequestParam(required = false, defaultValue = ApplicationConstants.DEFAULT_OFFSET + "") @Min(value = 0, message = "Offset parameter must be greater or equal 0") Integer offset) {
+        Map<String, String[]> reqParams = request.getParameterMap();
+        List<UserDto> users = userService.findAllForQuery(reqParams, limit, offset);
+        return ResponseEntity.ok(userAssembler.toCollectionModel(users, offset));
+    }
+
+
+    @GetMapping(path = "/{userId}/orders")
+    public ResponseEntity<CollectionModel<OrderDto>> getAllOrdersForUser(WebRequest webRequest, @PathVariable Long userId,
+                                                                         @RequestParam(required = false, defaultValue = ApplicationConstants.DEFAULT_LIMIT + "") @Min(value = 0, message = "Limit parameter must be greater or equal 0") Integer limit,
+                                                                         @RequestParam(required = false, defaultValue = ApplicationConstants.DEFAULT_OFFSET + "") @Min(value = 0, message = "Offset parameter must be greater or equal 0") Integer offset) {
+        Map<String, String[]> reqParamMap = new HashMap<>(webRequest.getParameterMap());
+        reqParamMap.put(ApplicationConstants.USER_ID_KEY, new String[]{String.valueOf(userId)});
+        List<OrderDto> orders = orderService.findAllForQuery(reqParamMap, limit, offset);
+
+        CollectionModel<OrderDto> order = CollectionModel.of(orders);
+        order.add(linkTo(methodOn(UserController.class)
+                .getAllOrdersForUser(null, userId, ApplicationConstants.DEFAULT_LIMIT, 0))
+                .withRel(ApplicationConstants.FIRST_PAGE));
+        order.add(linkTo(methodOn(UserController.class)
+                .getAllOrdersForUser(null, userId, ApplicationConstants.DEFAULT_LIMIT, offset + ApplicationConstants.DEFAULT_LIMIT))
+                .withRel(ApplicationConstants.NEXT_PAGE));
+
+        return ResponseEntity.ok(order);
+    }
+
+    @PostMapping(path = "/{userId}/orders")
+    public ResponseEntity<OrderDto> createOrder(@PathVariable Long userId,
+                                                @RequestBody @Valid List<SaveOrderDto> saveOrderDtoList) {
```

Comment:
Why did you create an additional SaveOrderDto class?
Can't you just use OrderDto as an input parameter?

Reply:
In my version of the application, client can choose several certificates of the same type. Therefore in order to avoid repetitions of similar certificates in request body, i made an additional DTO by which the cliend can list certificates and their number for creating an order.

Reply:
The thing is that you give a client an opportunity to buy several identical certificates, but what about buying different certificates in one order?

Reply:
Cliend can order several different certificates in one request by passing an array of SaveOrderDtos in a request body. SaveOrderDto(with 2 fields - certificateId and count of certificates with the id) is used only for getting rid of repetitions of passing the same certificateId several times in one request body. Example:
![image](https://user-images.githubusercontent.com/58271266/116853972-97663480-abff-11eb-8871-116a92780b82.png)

In this request body, the client make an attempt to order 2 certificates with id = 3 and 5 certificates with id = 4. All the certificates get into new created order(if there will not be exceptions).

Reply:
Ok then